code/components/extra-natives-five/src/VehicleExtraNatives.cpp

```diff
@@ -363,4 +367,110 @@ static HookFunction initFunction([]()
 	{
 		*reinterpret_cast<float *>(wheelAddr + WheelXOffsetOffset) = context.GetArgument<float>(2);
 	}));
+
+	static struct : jitasm::Frontend {
+		static bool ShouldSkipRepairFunc(fwEntity* VehPointer)
+		{
+
+			for (auto it = vehicles.begin(); it != vehicles.end(); it++) {
+
+				if (VehPointer == *it) {
+					return true;
+				}
+			}
+			return false;
+		}
+		virtual void InternalMain() override
+		{
+			//Restore registers from call and push registers to the stack to be preserved
+			mov(rcx, rbx);
+			push(rax);
+			push(rdx);
+
+			sub(rsp, 0x28);
+			mov(rax, (uintptr_t)ShouldSkipRepairFunc);
+			call(rax);
+			add(rsp, 0x28);
+
+			pop(rdx);
+			cmp(al, true); //al is 1 if repair func should be skipped
+			je("skiprepair");
+			pop(rax);
+			sub(rsp, 0x28);
+			AppendInstr(jitasm::InstrID::I_CALL, 0xFF, 0, jitasm::Imm8(2), qword_ptr[rax + 0x5D0]);
+			add(rsp, 0x28);
+			ret();
+			L("skiprepair");
+			pop(rax);
+			ret();
+		}
+	} asmfunc;
+
+	static auto * repairFunc = hook::pattern("48 8B 03 45 33 C0 B2 01 48 8B CB FF 90 D0 05 00 00 48 8B 4B 20").get_first(11);
+	static auto * vehicleDeconstructor = hook::pattern("41 BE 00 00 80 BF 48 89").get_first(-0x1B);
+	hook::nop(vehicleDeconstructor, 0x6);
+	hook::nop(repairFunc, 6);
+	hook::call_rcx(repairFunc, asmfunc.GetCode());
+
+	static struct : jitasm::Frontend {
+
+		static void LogStub(fwEntity* VehPointer)
+		{
+			auto count = vehicles.erase(VehPointer);
+			if (count > 0) {
+			}
+		}
+		virtual void InternalMain() override
+		{
+			mov(rcx, rax);
+			mov(r8d, r15d);
+
+			mov(rdi, rcx);
+			push(rax);
```

Comment:
why save _all_ these registers, aren't some of them callee-saved?

Reply:
https://docs.microsoft.com/en-us/cpp/build/caller-callee-saved-registers

Reply:
even then, look at the function to see what is in use and what isn't.